pypesto/collections/collections.py

```diff
@@ -0,0 +1,484 @@
+import numpy as np
+import pandas as pd
+from typing import Sequence, Tuple, Callable, Dict
+
+from ..prediction import PredictionResult, PredictionConditionResult
+from .constants import (PREDICTOR, PREDICTION_ID, PREDICTION_RESULTS,
+                        PREDICTION_ARRAYS, PREDICTION_SUMMARY, OUTPUT,
+                        OUTPUT_SENSI, TIMEPOINTS, X_VECTOR, NX, X_NAMES,
+                        NVECTORS, VECTOR_TAGS, PREDICTIONS, MODE_FUN,
+                        CollectionType, COLLECTION_TYPE, MEAN, MEDIAN,
+                        STANDARD_DEVIATION, PERCENTILE, SUMMARY, LOWER_BOUND,
+                        UPPER_BOUND)
+
+
+class CollectionPrediction:
+    """
+    A collection prediction consists of a collection, i.e., a set of parameter
+    vectors and their identifiers such as a sample or an ensemble, and a
+    prediction function. It can be attached to a collection-type object
+    """
+
+    def __init__(self,
+                 predictor: Callable,
+                 prediction_id: str = None,
+                 prediction_results: Sequence[PredictionResult] = None,
+                 lower_bound: Sequence[np.ndarray] = None,
+                 upper_bound: Sequence[np.ndarray] = None):
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        predictor:
+            Prediction function, e.g., an AmiciPredictor
+        prediction_id:
+            Identifier for the predictions
+        prediction_results:
+            List of Prediction results
+        lower_bound:
+            array of potential lower bounds for the parameters, should have the
+            same shape as the output of the predictions, i.e., a list of numpy
+            array (one list entry per condition), with the arrays having the
+            shape of n_timpoints x n_outputs for each condition
```

Comment:
Ah... no, the idea is: Add bounds to your prediction, to specify when you would say that a specific prediction is "non-identifiable". That's kinda analogous to parameters, but for model predictions... That's at least what it's meant for. According functionalities, examples, or checks are not yet implemented, but it will be useful in some not too long future...